crates/packaging/src/https.rs

```diff
@@ -102,6 +120,92 @@ impl<'a> PackageMetadata<'a> {
     }
 }
 
+#[test]
+#[should_panic(expected = "MissingHttps")]
+fn url_problem_missing_https() {
+    PackageMetadata::try_from("http://example.com").unwrap();
+}
+
+#[test]
+#[should_panic(expected = "MisleadingCharacter")]
+fn url_problem_misleading_character_at() {
+    PackageMetadata::try_from("https://user:password@example.com/").unwrap();
+}
+
+#[test]
+#[should_panic(expected = "MisleadingCharacter")]
+fn url_problem_misleading_character_unicode_2044() {
+    PackageMetadata::try_from("https://example.com⁄path").unwrap();
+}
+
+#[test]
+#[should_panic(expected = "MisleadingCharacter")]
+fn url_problem_misleading_character_unicode_2215() {
+    PackageMetadata::try_from("https://example.com∕path").unwrap();
+}
+
+#[test]
+#[should_panic(expected = "MisleadingCharacter")]
+fn url_problem_misleading_character_unicode_ff0f() {
+    PackageMetadata::try_from("https://example.com／path").unwrap();
+}
+
+#[test]
+#[should_panic(expected = "MisleadingCharacter")]
+fn url_problem_misleading_character_unicode_29f8() {
+    PackageMetadata::try_from("https://example.com⧸path").unwrap();
+}
+
+#[test]
+#[should_panic(expected = "InvalidFragment")]
+fn url_problem_invalid_fragment_not_a_roc_file() {
+    PackageMetadata::try_from("https://example.com/#filename.sh").unwrap();
+}
+
+#[test]
+#[should_panic(expected = "InvalidFragment")]
+fn url_problem_invalid_fragment_empty_roc_filename() {
+    PackageMetadata::try_from("https://example.com/#.roc").unwrap();
+}
+
+#[test]
+#[should_panic(expected = "MissingTarExt")]
+fn url_problem_not_a_tar_url() {
+    PackageMetadata::try_from("https://example.com/filename.zip").unwrap();
+}
+
+#[test]
+#[should_panic(expected = "InvalidExtensionSuffix")]
+fn url_problem_invalid_tar_suffix() {
+    PackageMetadata::try_from("https://example.com/filename.tar.zip").unwrap();
+}
+
+#[test]
+#[should_panic(expected = "MissingHash")]
+fn url_problem_missing_hash() {
+    PackageMetadata::try_from("https://example.com/.tar.gz").unwrap();
+}
+
+#[test]
+fn url_without_fragment() {
+    let actual = PackageMetadata::try_from("https://example.com/path/hash.tar.gz").unwrap();
+    assert_eq!(PackageMetadata {
+        cache_subdir: "example.com/path",
+        content_hash: "hash",
+        root_module_filename: None,
+    }, actual);
```

Comment:
For this style I'd have the `assert_eq` compare to `Ok(PackageMetadata { ... })` - this way, if it ever fails, you'll get a nice diff between the expected `Ok` and the actual `Err` instead of a panic.

Reply:
Yes, `assert_eq!` ftw. Thank you :)